be/src/vec/data_types/serde/data_type_map_serde.cpp

```diff
@@ -280,6 +280,15 @@ Status DataTypeMapSerDe::deserialize_one_cell_from_json(IColumn& column, Slice&
         ++elem_deserialized;
     }
 
+    if (nested_key_column.size() != nested_val_column.size()) {
+        // nested key and value should always same size otherwise we should popback wrong data
```

Comment:
why not just return error status? Doing popback will lose some key/value.

Reply:
now we just make sure right kv for right data